### PR TITLE
[19.09] Allow users to switch toolsheds if main toolshed is unavailable

### PR DESCRIPTION
**Pull request by guerler**
This PR allows users to switch to a different toolshed if the main toolshed is not available. It also improves the error message shown if a toolshed server is unavailable. Finally it adds the new `category` search option to category selections.

### CODE REVIEW

File: client/galaxy/scripts/components/Toolshed/Categories.vue [hunk 1]
```diff
@@ -65,7 +65,7 @@ export default {
                 });
         },
         onCategory(category) {
-            this.$emit("onCategory", category);
+            this.$emit("onCategory", `category:${category}`);
```
Comment (martenson) [inline client/galaxy/scripts/components/Toolshed/Categories.vue]: You need to enclose the category in single quotes `'` for categories that contain e.g. space

Comment (martenson) [reply client/galaxy/scripts/components/Toolshed/Categories.vue]: `_parse_reserved_filters("climate category:'Climate Analysis' owner:'bjoern gruening' psy_maps")`

Comment (guerler) [reply client/galaxy/scripts/components/Toolshed/Categories.vue]: Cool, Ok.

**Comment by martenson** [comment]
thanks @guerler !

**Pull request merged**